std/assembly/uri.ts

```diff
@@ -0,0 +1,29 @@
+import { encode, decode, URI_UNSAFE, URL_UNSAFE } from "./util/uri";
+
+export function encodeURI(str: string): string {
+  var len = str.length;
+  if (!len) return str;
+  var result = __new(len << 1, idof<String>());
```

Comment:
Would it be feasible to compute the final length here, so there's only one allocation? Should be cheaper I think.

Reply:
It could be cheaper for short strings but slower for long strings, so I'm not sure. It also increase code complexity